### PR TITLE
#164139646 Implement social authentication for google and facebook

### PR DESCRIPTION
**Pull request by codinger41**
#### What does this PR do?
It implements social authentication feature for Google and facebook.

#### Description of Task to be completed?
On the Login route, users should also be able to  Login via their favourite social media platform.

#### How should this be manually tested?
- Hit the endpoint for social authentication on the backend, ie `HOST/api/v1/auth/google`, it'll redirect the user to the service content page, then if the user accepts the consent messages then it redirects the user back to our frontend.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139646

#### Questions:

### CODE REVIEW

**Review by olajide1234 — DISMISSED** [review]

**Review by kleva-j — DISMISSED** [review]

File: src/components/header/Header.jsx [hunk 1]
```diff
@@ -30,6 +37,7 @@ const Header = (props) => {
                 <div>
                   <Link to="/" className={className}>Home</Link>
                   <Link to="/create-article" className={className}>Create post</Link>
+                  <a href="#" onClick={logout} className={className}>Log out</a>
```
Comment (9jaswag) [inline src/components/header/Header.jsx]: @leksyib why use an anchor tag here?

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: src/components/header/Header.jsx (hunk 1 above)
Comment (codinger41) [reply src/components/header/Header.jsx]: @9jaswag  <Link/> tag requires a `to` prop. So I decided to use the anchor tag. I could also use a <Link/> tag but before the component can know the route it's in, we'll have to pass `history` prop from every other component using the Header component.

**Comment by codinger41** [comment]
@9jaswag  fixed

File: src/components/header/Header.jsx (hunk 1 above)
Comment (9jaswag) [reply src/components/header/Header.jsx]: @leksyib what about just using a button?

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**